docs/resources/policy_v2.md

```diff
@@ -0,0 +1,156 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "cyral_policy_v2 Resource - terraform-provider-cyral"
+subcategory: ""
+description: |-
+    This resource allows management of various types of policies in the Cyral platform. Policies can be used to define access controls, data governance rules, and approval mechanisms to ensure compliance and security within your database environment.
+---
+
+# cyral_policy_v2 (Resource)
+
+This resource allows management of various types of policies in the Cyral platform. Policies can be used to define access controls, data governance rules, and approval mechanisms to ensure compliance and security within your database environment.
+
+## Example Usage
+
+```terraform
+resource "cyral_policy_v2" "local_policy_example" {
+  name        = "local_policy"
+  description = "Local policy to allow gym users to read their own data"
+  enabled     = true
+  tags        = ["gym", "local"]
+  scope {
+    repo_ids = ["2gaWEAyeKbETyUy1LSx985gVqrk"]
+  }
+  document    = jsonencode({
+    governedData = {
+      locations = ["gym_db.users"]
+    }
+    readRules = [
+      {
+        conditions = [
+          {
+            attribute = "identity.userGroups"
+            operator  = "contains"
+            value     = "users"
+          }
+        ]
+        constraints = {
+          datasetRewrite = "SELECT * FROM $${dataset} WHERE email = '$${identity.endUserEmail}'"
+        }
+      }
+    ]
+  })
+  enforced    = true
+  type        = "local"
+}
+
+resource "cyral_policy_v2" "global_policy_example" {
+  name        = "global_policy"
+  description = "Global policy for finance users with row limit for PII data"
+  enabled     = true
+  tags        = ["finance", "global"]
+  scope {
+    repo_ids = ["2gaWEAyeKbETyUy1LSx985gVqrk"]
+  }
+  document    = jsonencode({
+    governedData = {
+      tags = ["PII"]
+    }
+    readRules = [
+      {
+        conditions = [
+          {
+            attribute = "identity.userGroups"
+            operator  = "contains"
+            value     = "finance"
+          }
+        ]
+        constraints = {
+          maxRows = 5
+        }
+      },
+      {
+        conditions = []
+        constraints = {}
+      }
+    ]
+  })
+  enforced    = true
+  type        = "global"
+}
+
+
+resource "cyral_policy_v2" "approval_policy_example" {
```

Comment:
Approval policies are currently "invisible" - also proposed to be not documented to begin with.

Let's have a brief discussion in the next meeting on whether it's desirable to expose this construct in the docs (I don't have any mortal objection against this but we just need to be consistent - if we have them here, the documentation should also cover it).

Reply:
fixed in 78331719a89bf2f57045575781c918ae0141e7bd